packages/authentication/cardstack/middleware.js

```diff
@@ -164,15 +166,23 @@ class Authentication {
     if (!user.data || !user.data.type) { return; }
 
     let have;
+    let query = source['userCorrelationQuery'];
 
-    if (user.data.id != null) {
-      try {
+    try {
+      if (user.data.id != null) {
         have = await this.userSearcher.get(user.data.type, user.data.id);
```

Comment:
I don't think this should take priority over the userCorrelationQuery. It's legit to want to have externally -provided IDs while also doing custom correlation.

I think you can get rid of this step entirely by making the default userCorrelationQuery be one that just finds by ID.

Reply:
ah--good idea. will do